### PR TITLE
Contribution SLA

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Status
Ready for review

## Related Issues
fixes: [CIAC-4695](https://jira-hq.paloaltonetworks.local/browse/CIAC-4695).

## Description
- Added an SLA document for the contribution process.
- Added a few minor changes in the existing docs.

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Hi @anara123, @ShahafBenYakir, 
I added an initial version of a contribution SLA doc -[Contribution SLA](https://xsoar-pan-dev--pull-request-1253-1r1wk7xa.web.app/docs/contributing/sla).
The wording needs to be improved obviously, I'll ask for a tech doc review once we are done. 

Would appreciate your review and thoughts:
- Please let me know if you think something is missing or needs to be removed. 
- Please let me know what you think about the **time commitments** I defined.
- Do you think we should add a clause at the end about what they can do if our SLA commitments are not met? 


@dansterenson - Please review the change in [Contributing via Cortex XSOAR](https://xsoar-pan-dev--pull-request-1253-1r1wk7xa.web.app/docs/contributing/marketplace). Let me know if you think we need to add something to that page (maybe something regarding our last changes with contributing to existing packs or to API modules). 

Thanks in advance :)

**Review by dansterenson — APPROVED** [review]
[Contributing via Cortex XSOAR](https://xsoar-pan-dev--pull-request-1253-1r1wk7xa.web.app/docs/contributing/marketplace) section looks good to me.

File: docs/contributing/sla.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+id: sla
+title: Contribution SLA
+---
+
+This is a Service Level Agreement (SLA) between the contributor and the Cortex XSOAR content team. This document identifies the services required and the expected level of services.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:6]:
```suggestion
This is a Service Level Agreement (SLA) between the contributor and the Cortex XSOAR Content team. This document identifies the services required and the expected level of services.
```

File: docs/contributing/sla.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+In each of the above options, the review phase will begin with the **opening of the GitHub Pull Request** containing your content changes.
+
+#### Once your PR is open, the XSOAR content team commits to the following:
+
+   1. A reviewer will be assigned to your PR within an **hour** after opening it.
+   
+   2. Your reviewer will publish his initial response to your changes within **5 business days**.
+   
+   3. If you were required to make changes to the code, you have made the necessary changes, and have updated about it in a message on the PR - your reviewer will respond within **3 business days**.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:25]:
```suggestion
   3. If requested, after the needed changes were done, and a corresponding message in the pull request was posted - your reviewer will respond within **3 business days**. Note, you might have a couple of such fixes rounds, the above holds for each of them.
```

File: docs/contributing/sla.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+   
+   4. Your reviewer will be available for any questions during the review process - you can contact him on the PR itself or on slack (DFIR community).
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:27]: Add a link to the DFIR registration

File: docs/contributing/sla.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+   
+   5. Once your PR is approved and merged by your reviewer, an **internal PR** including your changes will be opened within an **hour**.
+    The internal PR allows us to run our internal validity and security checks on your final code. 
+   
+   6. Once the **internal PR is merged** your changes will be published in the XSOAR marketplace within **3 business days**.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:32]: As we don't commit to any timeline regarding merging the internal PR I don't think we should mention it, instead IMO we should only say that once their PR is merged the changes will be published in 3 business days.

File: docs/contributing/sla.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+
+#### In order for the contribution process to be successful, the XSOAR content team requires the contributors the following:
+
+   1. Provide the XSOAR content team with as much information as possible about the changes you made or about the new content you created.
+        * If you contributed through the XSOAR UI, please provide this information in the redirected **UI contribution form**.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:39]:
```suggestion
        * If you contributed through the XSOAR UI, please provide this information in your **UI contribution form**.
```

File: docs/contributing/sla.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      ![CLA](/doc_imgs/contributing/failed_CLA.png)
+      
+      
+   3. Providing the XSOAR content team with a recorded demo session that demonstrates your changes is much appreciated and will speed up the review process.
+        * If you contributed through the XSOAR UI, please provide a link to the recorded session in the redirected UI contribution form.
+        * If you contributed through a GitHub Pull Request, please provide it in the **contribution registration form**.
+   
+   4. Please check the status of the build of your PR once it is completed - If the build includes errors, please try to solve them - for more information about the build process please see [the-build-process](../contributing/conventions#the-build-process).
+   
+   5. During the review process monitor your PR - your reviewer will add comments to the Pull Request, asking questions and requesting changes. In order to establish a decent release process for your contribution, you are kindly asked to respond to the reviewer's code review and apply the required changes within **14 days**.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:58]:
```suggestion
   5. During the review process monitor your PR - your reviewer will add comments to the Pull Request, asking questions and requesting changes. In order to establish a decent review process for your contribution, you are kindly asked to respond to the reviewer's code review and apply the required changes within **14 days**.
```

File: docs/contributing/sla.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      **Stale Pull Requests might be closed**.
+    
+   6. In some cases, your reviewer will ask you to schedule a meeting to see an interactive demo. Make sure you have a working installation of Cortex XSOAR with your pack fully configured. Check out our [Contribution Demo Page](../contributing/demo-prep) for more details.
```
Comment (ShahafBenYakir) [inline docs/contributing/sla.md:61]:
```suggestion
   6. In some cases, your reviewer will ask you to schedule a meeting to see an interactive demo. Make sure you have a working installation of Cortex XSOAR with your most recent pack version (including all review comments) fully configured. Check out our [Contribution Demo Page](../contributing/demo-prep) for more details.
```

**Review by ShahafBenYakir — COMMENTED** [review]
Where will these be located? Which article will point to this?

File: docs/contributing/sla.md (hunk 3 above)
Comment (ShacharKidor) [reply docs/contributing/sla.md:27]: Good idea - Thanks! :)

**Comment by ShacharKidor** [comment]
> Where will these be located? Which article will point to this?

@ShahafBenYakir - I think I'll add a link to the SLA doc from the first page of the xsaor.pan.dev 'contributing' section. 
Maybe in the first paragraph or under the 'How to Contribute' header. WDYT?

File: docs/contributing/sla.md (hunk 4 above)
Comment (ShahafBenYakir) [reply docs/contributing/sla.md:32]: They will not as you tell them it will take up to 3 days

Comment (ShacharKidor) [reply docs/contributing/sla.md:32]: 3 business days after the merge of the internal PR - I'm trying to emphasize here that the tests we run on the internal PR are a step without a time commitment from our side.

**Review by richardbluestone — APPROVED** [review]

**Comment by MosheEichler** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed in 7 days if no further activity occurs.
> 
> If you'd like to keep it open, please leave a comment with the status of the PR.
> 
> Thank you for your contribution!

Removing, still relevant.

**Review by richardbluestone — APPROVED** [review]

**Pull request merged**